src/CommunityToolkit/Xamarin.CommunityToolkit/PlatformConfiguration/Multiplatform/StatusBar/StatusBar.ios.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using Foundation;
+using UIKit;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.iOS;
+
+namespace Xamarin.CommunityToolkit.PlatformConfiguration.Multiplatform
+{
+	public static partial class StatusBar
+	{
+		static partial void PlatformSetColor(Color color)
+		{
+			var uiColor = color.ToUIColor();
+
+			if (UIDevice.CurrentDevice.CheckSystemVersion(13, 0))
+			{
+				foreach (var window in UIApplication.SharedApplication.Windows)
+				{
+					const int statusBarTag = 38482;
+					var statusBar = window.ViewWithTag(statusBarTag) ?? new UIView(UIApplication.SharedApplication.StatusBarFrame);
+					statusBar.Tag = statusBarTag;
+					statusBar.BackgroundColor = uiColor;
+					statusBar.TintColor = uiColor;
+					window.AddSubview(statusBar);
+				}
+			}
+			else
+			{
+				var statusBar = UIApplication.SharedApplication.ValueForKey(new NSString("statusBar")) as UIView;
+				if (statusBar != null && statusBar.RespondsToSelector(new ObjCRuntime.Selector("setBackgroundColor:")))
+				{
+					statusBar.BackgroundColor = uiColor;
+				}
+			}
+
+			GetCurrentViewController().SetNeedsStatusBarAppearanceUpdate();
+		}
+
+		static partial void PlatformSetStyle(StatusBarStyle style)
+		{
+			var uiStyle = style switch
+			{
+				StatusBarStyle.LightContent => UIStatusBarStyle.LightContent,
+				StatusBarStyle.DarkContent => UIStatusBarStyle.DarkContent,
+				_ => UIStatusBarStyle.Default
+			};
+			UIApplication.SharedApplication.SetStatusBarStyle(uiStyle, false);
+
+			GetCurrentViewController().SetNeedsStatusBarAppearanceUpdate();
+		}
+
+		static UIViewController GetCurrentViewController()
+		{
+			var window = UIApplication.SharedApplication.KeyWindow;
+
+			var vc = window.RootViewController ?? throw new NullReferenceException();
+			while (vc.PresentedViewController != null)
```

Comment:
can we be in a deadlock here? Just imagining some edge case

Reply:
Do you mean infinite loop? Not sure since I'm not familiar with iOS development.
I found this on SO: https://stackoverflow.com/a/61642583/4312132
From what I understand it `RootViewController` returns controller that is currently presenting something. Not sure, why do we need a loop there.

Could you please debug this code and see what does `RootViewController` return and if we go inside the loop at all?

Reply:
Here's why do we need a loop here: "There is no single "current" view controller in iOS. There could be multiple view controllers on the screen at once (either nested or presented as popovers or modals). If we have modal page open, we need to use it's controller instead of one from the `window`. We need to find the "most-presented" (for lack of a better term) view controller. "

There shouldn't be an infinite loop here since we would exit when top-most controller is found.